src/nodedef.cpp

```diff
@@ -907,8 +907,15 @@ void ContentFeatures::updateTextures(ITextureSource *tsrc, IShaderSource *shdsrc
 	u32 overlay_shader = shdsrc->getShader("nodes_shader", overlay_material, drawtype);
 
 	// minimap pixel color = average color of top tile
-	if (tsettings.enable_minimap && !tdef[0].name.empty() && drawtype != NDT_AIRLIKE)
-		minimap_color = tsrc->getTextureAverageColor(tdef[0].name);
+	if (tsettings.enable_minimap && drawtype != NDT_AIRLIKE && !tdef[0].name.empty())
+	{
+		if (!tdef_overlay[0].name.empty()) {
+			std::string combined = tdef[0].name + "^(" + tdef_overlay[0].name + ")";
```

Comment:
This notation might not be immediately obvious, thus I would prefer a brief comment to help newcomers.
```suggestion
			// Use texture modifier to take tile overlays into account.
			std::string combined = tdef[0].name + "^(" + tdef_overlay[0].name + ")";
```